### PR TITLE
feat: audit -> cargo deny for pr/push and audit cron

### LINKED ISSUE #22: Add `cargo audit` action
Whilst https://github.com/fission-codes/rust-template/issues/21 mentions `cargo deny` - 

`cargo audit` action would be good to have as well in "cron basis"

Like this: https://github.com/libp2p/rust-libp2p/blob/master/.github/workflows/cargo-audit.yml

This is because `cargo deny` action doesn't raise issues in the repo for anything found in the background.

`cargo audit` action can be used to raise the issues in-repo for any advisories encountered

Personally I'm biased since I maintain a fork of it: https://github.com/rustsec/audit-check :)

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
# Description

Fixes #22 and #21 by adding cargo deny (instead of audit) for prs/etc and uses audit cron for schedule sec. checks

- [X] New feature (non-breaking change that adds functionality)

### CODE REVIEW

**Review by matheus23 — APPROVED** [review]
I don't have a lot of background when it comes to these things, but it just looks reasonable? :+1: 

I think you're doing this in your free time @pinkforest, but I think you may be someone much more qualified for reviewing this small PR. If you happen to have some time we'd appreciate it! :pray:

**Review by pinkforest — APPROVED** [review]
@matheus23 thanks for the kind words - the chore stuff is where the fun is at :sweat_smile: 

lgtm :partying_face: :tada:

**Comment by appcypher** [comment]
Finally learning about `cargo-deny`. It looks more flexible and goes beyond sec audits. Cool stuff.
This PR mostly looks good to me just one nit.

File: rust+wasm/deny.toml [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# This template contains all of the possible sections and their default values
```
Comment (appcypher) [inline rust+wasm/deny.toml:1]: What do you think about stripping the doc? It makes it hard to parse the actual config. I'm assuming the info here can be found on the doc site.

Comment (zeeshanlakhani) [reply rust+wasm/deny.toml:1]: @appcypher I thought it may be helpful for those editing it (as this is just the template). I saw that other "bigger" libs like linkered still had it, as most people run `cargo deny init` and then modify as it goes.

**Review by appcypher — APPROVED** [review]
PR looks good to me 🎉

File: rust+wasm/deny.toml (hunk 1 above)
Comment (pinkforest) [reply rust+wasm/deny.toml:1]: fwiw - when I am spinning up a new rust project the doc speeds me up considerably - reminding myself what was what - this is why it gets generated via `cargo deny init` :)
